firmware/src/hal/include/hal/nvmem.h

```diff
@@ -73,8 +72,10 @@ void nvmem_init();
  * @param key a string key to uniquely identify the block
  * @param addr the base address of the block
  * @param size the size of the block in bytes
+ *
+ * @return whether the block was successfully registered
  */
-void nvmem_register_block(char* key, void* addr, size_t size);
+bool nvmem_register_block(char* key, void* addr, uint8_t size);
```

Comment:
When we load the blocks from the secret store using `sest_read`, we verify if the secret was correctly read by doing:
```
if (sest_read(key, tmp, nvm_blocks[i].size) == nvm_blocks[i].size) {
    memcpy((uint8_t*)nvm_blocks[i].addr, tmp, nvm_blocks[i].size);
} else {
    LOG("Error loading NVM block <%s>\n", nvm_blocks[i].key);
    clear_blocks();
    free(key);
    return false;
}
```

but `sest_read` returns a `uint8_t`. So, if we allow a block size to be over `255`, this check will never succeed, even if we manage to read it correctly. I see two possible fixes to this: limiting the block size to a `uint8_t` to prevent this inconsistency or having `sest_read` return a `size_t` instead. I'm good with both of them, but just limiting the block size here is the path of least resistance, since the return of `sest_read` is currently treated as a `uint8_t` by the other callers (seed and access) , and our secrets are under this limit. What do you think?

Reply:
I see the problem, but it is a known issue. If we happen to ever register blocks bigger than 255 bytes, then the app will stop working. But the NVM module should allow it, and we should - in the future - fix the secret store (and key value store) to allow for bigger value sizes. It is unclear why the compiler is not warning about not casting on the last parameter of e.g. `sest_read(key, tmp, nvm_blocks[i].size)`. Do you know why?

Reply:
Got it. In this case, I'll revert to `size_t` and keep the tests limited to 255 bytes with a comment to remind us this is a known limitation. How does that sound?

Regarding the compiler not complaining, I don't know the reason. Will double check the flags we are using, maybe we're not enabling some warnings for the sgx targets

Reply:
Sounds good. Please check the flags and add them if necessary.

Reply:
Reverted back to `size_t`

Reply:
Regarding the flags, I double-checked and we are enabling all warnings (and also `-Werror`) for the sgx build. My guess is that the compiler didn't complain because the implicit cast was casting a `uint8_t` to a `size_t`, which is fine, the warning would be present if it was the other way around.

In any case, this reminded me of adding these flags to the tests as well just to make sure we keep the tests warning-free, so I added them and fixed some warnings on `mock_secret_store.c`.